torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -4,7 +4,9 @@
 #include <c10/util/env.h>
 
 #ifdef USE_C10D_NCCL
+#include <vector>
 
+#include <cuda_runtime.h>
```

Comment:
Move to NCCLUtils since we need to gate it with Macro so that builds in other environment are happy.